components/button/button-add.js

```diff
@@ -69,6 +83,34 @@ class ButtonAdd extends PropertyRequiredMixin(FocusMixin(LocalizeCoreElement(Lit
 			:host([mode="icon-when-interacted"]) button:not(:focus):not(:hover) d2l-button-add-icon-text {
 				position: absolute;
 			}
+			:host([mode="icon-when-interacted"]) button:hover d2l-button-add-icon-text,
+			:host([mode="icon-when-interacted"]) button:focus d2l-button-add-icon-text {
+				animation: position-change-animation 50ms step-end 0ms 1 forwards; /* add delay in changing position to avoid flash of missing icon space */
+			}
+
+			@media (prefers-reduced-motion: reduce) {
+				button:hover .line,
+				button:focus .line {
+					animation: none !important;
+					background: var(--d2l-color-celestine-minus-1);
+				}
+				:host([mode="icon-when-interacted"]) button:hover .line {
+					transition: background 100ms step-end;
+				}
+			}
+
+			@keyframes line-start-animation {
+				0% { background: linear-gradient(to right, var(--d2l-color-mica) 0%, var(--d2l-color-mica) 11%, var(--d2l-color-celestine-minus-1) 11%) left center / 135%; }
```

Comment:
This is supposed to be a gradient animation that starts at around 85% length of the line outwards on each side and then fills to 100%.